app/models/base_facility.rb

```diff
@@ -167,27 +167,38 @@ def radial_query(params)
         Float(params[:long]),
         params[:type],
         params[:services],
+        params[:ids],
         limit
       )
     end
 
-    def build_distance_result_set(lat, long, type, services, limit = nil)
+    # rubocop:disable Metrics/MethodLength
+    # rubocop:disable Metrics/ParameterLists
+    def build_distance_result_set(lat, long, type, services, ids, limit = nil)
       additional_data = <<-SQL
```

Comment:
you can probably fix the method length by pulling the additional_data query into a method instead of in-lining, fwiw